tmail-backend/webadmin/webadmin-rate-limit/src/main/java/com/linagora/tmail/webadmin/RateLimitPlanUserRoutes.java

```diff
@@ -0,0 +1,192 @@
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Route getPlanOfUser() {
+        return (request, response) -> {
+            Username username = extractUsername(request);
+            userPreconditions(username);
+            return Mono.from(planUserRepository.getPlanByUser(username))
+                .map(planId -> new RateLimitingPlanIdResponse(planId.value().toString()))
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .onErrorResume(RateLimitationPlanNotFoundException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.NOT_FOUND_404)
+                        .type(ErrorResponder.ErrorType.NOT_FOUND)
+                        .message(String.format("User %s does not have a plan", username.asString()))
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Route revokePlanOfUser() {
+        return (request, response) -> {
+            Username username = extractUsername(request);
+            userPreconditions(username);
+            return Mono.from(planUserRepository.revokePlan(username))
+                .then(Mono.just(Responses.returnNoContent(response)))
+                .onErrorResume(NullPointerException.class, e -> {
+                    throw ErrorResponder.builder()
+                        .statusCode(HttpStatus.BAD_REQUEST_400)
+                        .type(ErrorResponder.ErrorType.INVALID_ARGUMENT)
+                        .message(e.getMessage())
+                        .haltError();
+                })
+                .block();
+        };
+    }
+
+    private Username extractUsername(Request request) {
+        return Username.of(request.params(USERNAME_PARAM));
+    }
+
+    private RateLimitingPlanId extractPlanId(Request request) {
+        return new RateLimitingPlanId(UUID.fromString(request.params(PLAN_ID_PARAM)));
+    }
+
+    public void userPreconditions(Username username) throws UsersRepositoryException {
+        if (!usersRepository.contains(username)) {
+            throw ErrorResponder.builder()
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .type(ErrorResponder.ErrorType.NOT_FOUND)
+                .message(String.format("User %s does not exist", username.asString()))
+                .haltError();
+        }
+    }
+
+    public void planIdPreconditions(RateLimitingPlanId planId) {
+        if (Boolean.FALSE.equals(Mono.from(planRepository.planExists(planId)).block())) {
```

Comment:
```suggestion
        if (Mono.from(planRepository.planExists(planId)).block()) {
```

Is easier for the mere mortal I am. Our target audience is not the intelliJ compiler.